### PR TITLE
Update AbiIO to preserve bytes value

### PR DESCRIPTION
**Pull request by bow**
This pull request addresses issue #1982 and is a continuation from PR #1984.

@peterjc This PR incorporates all the changes we have discussed so far (including the fallback to system encoding when we can't extract string values for `SeqRecord.id` and `SeqRecord.description). There is also a test case included, with the abif file containing the non-ascii tag that @zqfang reported earlier.

- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/SeqIO/AbiIO.py [hunk 1]
```diff
@@ -331,6 +334,23 @@
...
+def _get_string_tag(opt_bytes_value, default=None):
+    """Return the string value of the given an optional raw bytes tag value.
+
+    If the bytes value is None, return the given default value.
+
+    """
+    if opt_bytes_value is None:
+        return default
+    try:
+        return _bytes_to_string(opt_bytes_value)
```
Comment (peterjc) [inline Bio/SeqIO/AbiIO.py]: Doesn't this use the default encoding? The NEWS entry says you'd try ASCII first...

Comment (bow) [reply Bio/SeqIO/AbiIO.py]: Hmm...I did say the first encoding attempt assumes `utf-8`, though I did say the parser can be made more robust by handling non-ascii characters in the beginning of the entry.

I'll revise the `NEWS.rst` entry to be clearer.

Comment (peterjc) [reply Bio/SeqIO/AbiIO.py]: I may be confused/wrong, but does ``.decode()`` not use the system default encoding anyway - which may not be utf-8?

Comment (bow) [reply Bio/SeqIO/AbiIO.py]: [Looking at the docs](https://docs.python.org/3/library/codecs.html#codecs.decode), I think it uses `utf-8` by default, regardless of the system encoding.

Initially I thought: is it also possible that the value shown in the docs is actually the result of `sys.getdefaultencoding()`? That seems unlikely however, and I feelthat  they would have said so had it it been the case.

Comment (peterjc) [reply Bio/SeqIO/AbiIO.py]: OK, you're right. Thanks. I've not looked at the encoding and locales for a while now.

**Comment by peterjc** [comment]
You need to tweak ``NEWS.rst`` to pass validation, but otherwise this looks good to merge.

File: Bio/SeqIO/AbiIO.py (hunk 1 above)
Comment (bow) [reply Bio/SeqIO/AbiIO.py]: Great, ok ~ the tests finally pass, so I am marking this as resolved and will merge this soon today :).

**Comment by bow** [comment]
Ok, it seems to be fine now. I'm rebasing this onto master now ~

**Pull request merged**